var/spack/packages/spot/package.py

```diff
@@ -0,0 +1,18 @@
+from spack import *
+import os
+
+class Spot(Package):
+    """Spot is a C++11 library for omega-automata manipulation and model checking."""
+    homepage = "https://spot.lrde.epita.fr/index.html"
+    url      = "http://www.lrde.epita.fr/dload/spot/spot-1.99.3.tar.gz"
+
+    version('1.99.3', 'd53adcb2d0fe7c69f45d4e595a58254e')
+
+    #depends_on("gcc@4.8:")
```

Comment:
So, in that example, you could restrict the boost you build ROSE with, without restricting the boost package in general.  In your example, the user hasn't overridden the compiler for boost on the command line, Spack will currently just build boost with whatever compiler ROSE used (dependencies inherit from the nearest parent with a defined compiler).  So unless they're explicit, you'll get a compatible compiler and the user can still experiment with others if he's adventurous.

Do you need more constraints than that?  If I put `require('gcc@4.7')` in the boost package, everyone has to build boost with gcc.